src/screens/SettingsScreen.tsx

```diff
@@ -42,20 +55,156 @@ export const SettingsScreen: React.FC<SettingsScreenProps> = () => {
   const { prefs, save } = usePrefs()
   const canUseLocalAuth = useCanUseLocalAuth()
   const { user, handleLogout } = useAuth()
+  const { secrets, reset, add } = useSecrets()
+  const [status, requestPermission] = MediaLibrary.usePermissions()
+
+  const backupFileNamePattern = 'optic-backup-{TS}'
+
+  const androidExport = async (fileName, fileContent) => {
+    const permissions =
+      await StorageAccessFramework.requestDirectoryPermissionsAsync()
+    if (!permissions.granted) {
+      Toast.show('Permissions not granted')
+      return
+    }
+
+    try {
+      const uri = await StorageAccessFramework.createFileAsync(
+        permissions.directoryUri,
+        fileName,
+        'application/text'
+      )
+
+      await writeAsStringAsync(uri, fileContent, {
+        encoding: EncodingType.UTF8,
+      })
+      Toast.show('Tokens exported successfully')
+    } catch (err) {
+      Toast.show('An error occurred while exporting tokens')
+      console.log(err)
+    }
+  }
+
+  const iosExport = async (fileName, fileContent) => {
+    const available = Sharing.isAvailableAsync()
+    if (!available) {
+      Toast.show('Sharing not available')
+      return
+    }
+
+    requestPermission()
+
+    if (!status.granted) {
```

Comment:
I reviewed better the requestPermission() from the MediaLibrary and we don't need it, because it relates to the media library and we are not working with videos or images files, in other words, we don't need access to the gallery. I tested it without asking for the permissions on iOS (with a brand new simulator) and it still works, because under the hood the expo-sharing library is not storing a file in a protected or reserved route.

Reply:
that would explain why this works then, as per my previous comment